yeon_seyeon/week05/mission01/src/components/LoginPage.jsx

```diff
@@ -0,0 +1,106 @@
+// src/components/LoginPage.jsx
+import React, { useState } from "react";
+import styled from "styled-components";
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  padding: 50px;
+  color: white;
+`;
+
+const Title = styled.h1`
+  margin-bottom: 20px;
+`;
+
+const Input = styled.input`
+  width: 100%;
+  max-width: 300px;
+  padding: 10px;
+  margin-bottom: 10px;
+  border: 1px solid #ddd;
+  border-radius: 5px;
+  &:focus {
+    border-color: #ff007f;
+  }
+`;
+
+const Button = styled.button`
+  width: 100%;
+  max-width: 300px;
+  padding: 10px;
+  background-color: #ff007f;
+  color: white;
+  border: none;
+  border-radius: 5px;
+  cursor: pointer;
+  &:hover {
+    background-color: #ff66a3;
+  }
+`;
+
+const ErrorMessage = styled.p`
+  color: red;
+  font-size: 14px;
+  margin: 5px 0;
+`;
+
+const LoginPage = () => {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [emailError, setEmailError] = useState("");
+  const [passwordError, setPasswordError] = useState("");
+
+  const handleEmailChange = (e) => {
+    setEmail(e.target.value);
+    const emailPattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    if (!emailPattern.test(e.target.value)) {
+      setEmailError("올바른 이메일 형식이 아닙니다. 다시 입력해주세요!");
+    } else {
+      setEmailError("");
+    }
+  };
```

Comment:
else는 요즘 잘 안쓴다구 하더라구여!! else 대신 early return 문으로 하시는걸 추천드려요!! 

```suggestion
      return setEmailError("올바른 이메일 형식이 아닙니다. 다시 입력해주세요!");
    } 
      setEmailError("");
   
  };
```